webapp-django/crashstats/crashstats/models.py

```diff
@@ -195,10 +195,23 @@ def get_cached_result(key, instance, stringified_args):
                     os.remove(cache_file)
                 else:
                     logger.debug("CACHE FILE HIT %s" % stringified_args)
-                    if instance.expect_json:
-                        return json.load(open(cache_file))
-                    else:
-                        return open(cache_file).read()
+                    delete_cache_file = False
+                    with open(cache_file) as f:
+                        if instance.expect_json:
+                            try:
+                                return json.load(f)
+                            except ValueError:
+                                logger.warn(
+                                    "%s is not a valid JSON file and will "
+                                    "be deleted" % (
+                                        cache_file,
+                                    )
+                                )
+                                delete_cache_file = True
+                        else:
+                            return f.read()
+                    if delete_cache_file:
+                        os.remove(cache_file)
```

Comment:
I suppose you had to delay the removal of this file because it can't be done while it's open, right?